### PR TITLE
#159965290 Upgrade Author's Haven app from Django 1.10 to Django2.1

### PR DESCRIPTION
**Pull request by micahoriaso**
#### What does this PR do?
- Upgrade Author's Haven app to Django 2.1 from Django 1.10

#### Description of Task to be completed?
- Upgrade Django from version 1.10 to version 2.1
- Upgrade Django Rest Framework to a version compatible with Django 2.1 - Upgraded to version 3.8.2
- Upgrade Django Extensions to a version compatible with Django 2.1 - Upgraded to version 2.1.2

#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Checkout into the `ch-upgrade-django1.10-to-django2.1-159965290` branch
- Create a virtual environment by running `python3 -m venv venv`
- Activate the virtual enviroment by running `source venv\bin\activate`
- Run `pip install -r requirements.txt`
- Run `django-admin version` It should show `2.1`

#### What are the relevant pivotal tracker stories?
[Upgrade Authors Haven Django app from Django 1.10 to Django 2.1](https://www.pivotaltracker.com/n/projects/2192364/stories/159965290)

### CODE REVIEW

**Comment by micahoriaso** [comment]
@pngg @Bileonaire  @mikenthiwa @PAbishai @meshack-mbuvi  kindly review this pull request

**Review by pngg — COMMENTED** [review]
I've noticed that you've used both ```re_path``` and ```path``` for routes in ```authentication``` and ```authors``` apps respectively. I recommend that we stick to either of the functions for consistency.

**Comment by pabishai** [comment]
@micahoriaso Consider using `django-admin version` since the aim is to simply verify the version of django

**Comment by micahoriaso** [comment]
@pngg I have replaced re_path with path

**Comment by micahoriaso** [comment]
@PAbishai  I have changed the final step in how to test this pull request manually to : run django-admin version to check for the django version installed

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]
Under settings.py file, update the following lines(comments) to reflect the new version of Django: lines
80 https://docs.djangoproject.com/en/1.11/ref/settings/#databases
108 https://docs.djangoproject.com/en/1.11/topics/i18n/
121 https://docs.djangoproject.com/en/1.10/howto/static-files/

**Comment by micahoriaso** [comment]
@meshack-mbuvi 
- The database settings have been worked on by @mikenthiwa as well as the internationalization settings
- At this point we are working on the backend hence there is no need for configuring the static files. We will hold on that point until we get to the frontend part

File: authors/apps/authentication/urls.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-from django.conf.urls import url
+from django.urls import path
 
 from .views import (
     LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
 )
```
Comment (mikenthiwa) [inline authors/apps/authentication/urls.py:6]: As a good practice, you can add app_name variable which indicates the name of the app

File: .gitignore [hunk 2]
```diff
@@ -91,3 +91,6 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+# vscode
+.vscode/
```
Comment (parseen254) [inline .gitignore]: newline at the end of your file please.

File: authors/apps/authentication/urls.py (hunk 1 above)
Comment (parseen254) [reply authors/apps/authentication/urls.py:6]: @micahoriaso implement this.

File: requirements.txt [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
 django-cors-middleware==1.2.0
-django-extensions==1.7.1
-djangorestframework==3.4.4
+django-extensions==2.1.2
+djangorestframework==3.8.2
+isort==4.3.4
+lazy-object-proxy==1.3.1
+mccabe==0.6.1
 PyJWT==1.4.2
+pylint==2.1.1
```
Comment (parseen254) [inline requirements.txt]: Make sure to include only necessary requirements in this file.... stuff like pylint should not be here.

**Review by parseen254 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (micahoriaso) [reply .gitignore]: Working on it

File: authors/apps/authentication/urls.py (hunk 1 above)
Comment (micahoriaso) [reply authors/apps/authentication/urls.py:6]: This has been duly implemented

File: .gitignore (hunk 2 above)
Comment (micahoriaso) [reply .gitignore]: This has been fixed

File: requirements.txt (hunk 3 above)
Comment (micahoriaso) [reply requirements.txt]: Unnecessary third party packages have been removed

**Comment by micahoriaso** [comment]
@meshack-mbuvi, @parseendavid  I have attended to your reviews. Kindly let me know if they are met. Thank you.

**Review by meshack-mbuvi — APPROVED** [review]

**Review by parseen254 — APPROVED** [review]

**Pull request merged**